### DISCUSSION TITLE
Performance issues cloning repo

### DISCUSSION THREAD

**Discussion by to-fr** [eclipse-jgit/jgit#80] [General]
Hi,

we have several big repos in our company, e.g. 800 MB Git, 8 GB LFS.

I am currently using Eclipse 2024-03 for cloning the repos with

Import -> Git -> Projects from Git (with smart import)
Clone via https.
Branch Selection: default (all)
Tag fetching strategy: default (When fetching a commit, also fetch its tags)

Eclipse:
Receiving objects (634.831) 1:00
Resolving deltas (201.830) 0-53% (15 sec), 53-100% 07:00min
Checking out files (3805) 0:40
Total ~8:55 min

Command Line:
Receiving objects (644.480) 0:50
Resolving deltas (206.003) 1:44
Updating files (3805) 0:19
Filterung content 0:04
Total ~2:57 min

JGit needs about 9 minutes for cloning, the command line 3 minutes.

Is there any way to get the resolving deltas faster?

**Comment by msohn** [comment]
You can try to increase JGit buffer cache size in ~/.gitconfig, e.g.:
```
core.packedGitLimit=1g      # JGit buffer cache size
core.deltaBaseCacheLimit=256m # delta base cache size
core.packedGitUseStrongRefs=true # prevent Java GC can drop the buffer cache
```
you may need to also increase max heap size for Eclipse in eclipse.ini

**Comment by msohn** [comment]
If this clone involves any LFS objects being transferred I wouldn't be surprised if that is slower.
Note: I don't use LFS.

**Comment by to-fr** [comment]
Thank you for your advice!
Increasing the buffer size has unfortunately no effect in decreasing the time "Resolving deltas" takes.
I don't know if "Resolving deltas" involves LFS object transfer.
Is "Resolving deltas" a multithreaded operation so that increasing the number of threads could be beneficial?
